cmd/package-operator-manager/main.go

```diff
@@ -30,7 +33,11 @@ const (
 var di *dig.Container
 
 func main() {
-	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+	opts := zap.Options{
+		Development: false,
+		Level:       zapcore.ErrorLevel,
+	}
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
```

Comment:
What are the implications of this change?

Reply:
This is to reduce the log level to Error by default

Reply:
Do you know how the numeric verbosity log levels and the named log level here interact with each other?

Reply:
This is an interesting topic, so after I tried to compare other operators, from the doc here: https://github.com/operator-framework/operator-sdk/blob/master/website/content/en/docs/building-operators/golang/references/logging.md#default-zap-logger, so the explaination:
```
--zap-log-level: Zap Level to configure the verbosity of logging. Can be one of 'debug', 'info', 'error', or any integer value > 0 which corresponds to custom debug levels of increasing verbosity")
```
So based on this, actually, `log.info` equals `log.V(1).info` when we assign the `--zap-log-level=1`. I am not sure whether we need to update all the log statements to `log.V(constants.LogLevelInfo).Info`:thinking:

Reply:
And this is a little complicated, for the further ticket PKO-243, people have options to set string-based log levels and also numeric ones
For example:
- `--zap-log-level=error` only log.error will be printed out
- `--zap-log-level=info/debug`, then log.info and log.V(n).info will be printed out as well as the errors 
- `--zap-log-level>=1` log.info and log.V(1).info will be printed out, as well as the errors, and so on

Reply:
My initial thought on this is that we directly use the controller manager's `--zap-log-level` option and do not need to create a customer option, what we will need in `components/options.go` is to take a new env value to set `--zap-log-level` if there is any. So this means, in PKO-243, pko manager can take a new config parameter like `logLevel`, and we can provide the value like `info`, and `error` as well as integers like 0-5. Here the error will be the default one which mentioned in the ticket. 
I am working on testing this approach, but I did meet the problem of making manifest to accept bother string and integer values. 
So based on your comments, it looks like you want new a custom option instead of using `--zap-log-level`. Let me know which approach you prefer.

Reply:
Okay, I made it work for using `--zap-log-level`, this is the draft PR https://github.com/package-operator/package-operator/pull/1965 for PKO-243 based on this PR's change. It is still in progress, but I can set different log levels to change PKO's log verbosity. 

> Also IIUC then logr always logs errors, regardless of the verbosity level. I don't know how this works with the zapr (zap<->logr) and zap<->controller-runtime adapters though.
I will provide my testing result and comment on the ticket later

Reply:
Hi @weherdh,
I have a question about this change. I tested this locally and it looks like it also works with just this single change (lines 36-40). Are all the other changes necessary (setting verbosity at each log call)? Am I missing something?

Reply:
Yes, to change the verbosity is what Josh had discussed with me, I shared some info with you about that